ndk-build/src/apk.rs

```diff
@@ -94,6 +96,25 @@ impl<'a> UnalignedApk<'a> {
         Ok(())
     }
 
+    pub fn add_runtime_libs(
+        &self,
+        path: &Path,
+        target: Target,
+        search_paths: &[&Path],
+    ) -> Result<(), NdkError> {
+        let abi_dir = path.join(target.android_abi());
+        if abi_dir.is_dir() {
+            for entry in fs::read_dir(abi_dir)? {
+                let entry = entry?;
+                let path = entry.path();
+                if path.extension() == Some(OsStr::new("so")) {
+                    self.add_lib_recursively(&path, target, search_paths)?;
+                }
+            }
```

Comment:
Is this supposed to not recurse? Libraries under `<path>/<abi>/*.so` are included, but nothing beyond that (ie. `<path>/<abi>/**/*.so`.

Reply:
It didn't occur to me that you could place dynamically loaded libraries under subfolders of the abi directory, but it sounds like a valid use case. I'll make it recurse then.

Reply:
Trying to implement this now, and noticing that `add_lib()` copies the library to a `lib/<abi>/*.so` file within a temporary folder, before calling `aapt`.

It means that the folder hierarchy ends up being flattened. If the same filename appears several time in different subfolders, the command fails on the second one because the file already exists in the apk.

I can't think of a simple way to handle this use-case without either introducing some additional complexity in `add_lib()` or some code duplication to bypass `add_lib()`.

I suggest not making this recursive for now, as I'm not even sure this would actually ever be used.

Reply:
`add_lib()` might be able to take care of this by simply passing the relative path instead of just the filename to get flattened hierarchy again (otherwise files obviously end up in the wrong/unexpected directory). It would have to create a directory for `.parent()` though, and I haven't thought out how to deal with recursively needed libraries, those should still be placed at the root.

Let's leave this for now, we'll get to it when someone has a usecase - and is consequently able to answer these questions and validate  the result.